rpc/src/v1/impls/eth.rs

```diff
@@ -995,4 +996,27 @@ impl<C, SN: ?Sized, S: ?Sized, M, EM, T: StateInfo + 'static> Eth for EthClient<
 	fn compile_solidity(&self, _: String) -> Result<Bytes> {
 		Err(errors::deprecated("Compilation of Solidity via RPC is deprecated".to_string()))
 	}
+
+    fn get_encryption_public_key(&self, receiver_address: RpcH160) -> Result<RpcH256> {
+		Err(errors::light_unimplemented(None))
+    }
+
+    fn encrypt(&self, encryption_public_key: RpcH256, version: String, data: Bytes, padding: Trailing<RpcU64>) -> Result<Bytes> {
+        let mut data_vec: Vec<u8> = data.into();
+        let mut message = data_vec.as_mut_slice();
+        let public_key_bytes: [u8; 32] = encryption_public_key.0;
+        let ad: [u8; 0] = [0; 0];
+        let mut nonce = vec![0; 12];
+        let rand = SystemRandom::new();
```

Comment:
Is there a way to use a deterministic source for tests or supply `nonce` externally for the function so that we can write some tests?